### PR TITLE
Add summary message for connected batteries with disconnected regulating terminals

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

File: src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java [hunk 1]
```diff
@@ -40,6 +40,23 @@
...
+        public final String conversionMessageKey;
+        public final String disconnectedTerminalMessageKey;
+        public final String summaryMessageKey;
+
+        ConversionEquipmentType(String equipmentsLabel, String conversionMessageKey, String disconnectedTerminalMessageKey, String summaryMessageKey) {
+            this.equipmentsLabel = equipmentsLabel;
+            this.conversionMessageKey = conversionMessageKey;
+            this.disconnectedTerminalMessageKey = disconnectedTerminalMessageKey;
+            this.summaryMessageKey = summaryMessageKey;
+        }
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java:57]: You can replace that with `@AllArgsConstructor`.

File: src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java [hunk 2]
```diff
@@ -55,8 +72,10 @@ protected ReportNode forShortCircuitAnalysis(@NonNull final ReportNode reportNod
             logVoltageLevelsWithWrongIpValues(reportNode, runContext);
         }
         reportNode.getChildren().forEach(child -> {
-            if (child.getMessageKey().equals("generatorConversion")) {
-                insertReportNode(newReporter, forGeneratorConversion(child));
+            if (child.getMessageKey().equals(ConversionEquipmentType.GENERATOR.conversionMessageKey)) {
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java:75]:
```suggestion
            if (ConversionEquipmentType.GENERATOR.conversionMessageKey.equals(child.getMessageKey())) {
```

File: src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java [hunk 3]
```diff
@@ -55,8 +72,10 @@ protected ReportNode forShortCircuitAnalysis(@NonNull final ReportNode reportNod
...
+                insertReportNode(newReporter, forEquipmentConversion(child, ConversionEquipmentType.GENERATOR));
+            } else if (child.getMessageKey().equals(ConversionEquipmentType.BATTERY.conversionMessageKey)) {
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java:77]:
```suggestion
            } else if (ConversionEquipmentType.BATTERY.conversionMessageKey.equals(child.getMessageKey())) {
```

File: src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java [hunk 4]
```diff
@@ -92,8 +111,8 @@ protected ReportNode forGeneratorConversion(@NonNull final ReportNode reportNode
         ReportNodeAdder logsRegulatingTerminalSummaryAdder = null;
         TypedValue logsRegulatingTerminalSeverity = null;
         for (final ReportNode child : reportNode.getChildren()) {
-            if (child.getMessageKey().equals("disconnectedTerminalGenerator")) {
-                //we match line "Regulating terminal of connected generator MY-NODE is disconnected. Regulation is disabled."
+            if (child.getMessageKey().equals(conversionEquipmentType.disconnectedTerminalMessageKey)) {
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java:114]:
```suggestion
            if (conversionEquipmentType.disconnectedTerminalMessageKey.equals(child.getMessageKey())) {
```

File: src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java [hunk 5]
```diff
@@ -76,11 +95,11 @@ protected void logVoltageLevelsWithWrongIpValues(ReportNode reportNode, ShortCir
     }
 
     /**
-     * Modify node with key {@code generatorConversion}
+     * Modify node with key equals to {@code ConversionEquipmentType.conversionMessageKey}
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java:98]:
```suggestion
     * Modify node with key equals to {@link ConversionEquipmentType#disconnectedTerminalMessageKey}
```

**Review by Tristan-WorkGH — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java (hunk 1 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java]: Done

File: src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java (hunk 3 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java]: Done

File: src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java (hunk 4 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java]: Done

File: src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java (hunk 5 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java]: Done

**Review by Tristan-WorkGH — APPROVED** [review]

**Pull request merged**